ListLibrary/MyList.cs

```diff
@@ -0,0 +1,590 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ListLibrary
+{
+    public class MyList<T> : IMyList<T> where T : IComparable<T>
+    {
+        private const int _defaultCapacity = 4;
+        private const double _capacityRise = 1.33;
+
+        private T[] _items;
+        private int _size;
+
+        static readonly T[] _emptyArray = new T[0];
+
+        #region Constructors, Propeerties and Interfaces implementation
+
+        public MyList()
+        {
+            _items = _emptyArray;
+        }
+
+        public MyList(int capacity)
+        {
+            if (capacity < 0)
+            {
+                throw new ArgumentException("Capacity should be >= 0");
+            }
+
+            if (capacity == 0)
+            {
+                _items = _emptyArray;
+
+            }
+            else
+            {
+                _items = new T[capacity];
+            }
+        }
+
+        public MyList(IEnumerable<T> collection)
+        {
+            if (collection == null)
+            {
+                throw new ArgumentException("Collection can't be null");
+            }
+
+            _items = _emptyArray;
+
+            foreach (T item in collection)
+            {
+                Add(item);
+            }
+        }
+
+        public int Count => _size;
+
+        public int Capacity
+        {
+            get
+            {
+                return _items.Length;
+            }
+            set
+            {
+                if (value < _size)
+                {
+                    throw new ArgumentOutOfRangeException("Capacity is too small");
+                }
+                else
+                {
+                    T[] newItems = new T[value];
+
+                    if (_size > 0)
+                    {
+                        Array.Copy(_items, newItems, _size);
+                    }
+
+                    _items = newItems;
+
+                    if (value < 0)
+                    {
+                        _items = _emptyArray;
+                    }
+                }
+            }
+        }
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            for (int i = 0; i < _size; i++)
+            {
+                yield return _items[i];
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        private void EnsureCapacity(int min)
```

Comment:
Private methods should be moved to the end of the file

Reply:
Title is a bit invalid. Maybe `CalculateCapacity` or `UpdateCapacity`

Reply:
fixed